dev/system-test/firestore.ts

```diff
@@ -913,6 +914,37 @@ describe('DocumentReference class', () => {
         maybeRun();
       });
     });
+
+    it('handles more than 100 concurrent listeners', async () => {
+      const ref = randomCol.doc('doc');
+
+      const emptyResults: Array<Deferred<void>> = [];
+      const documentResults: Array<Deferred<void>> = [];
+      const unsubscribeCallbacks: Array<() => void> = [];
+
+      // A single GAPIC client can only handle 100 concurrent streams. We set
+      // up 100+ long-lived listeners to verify that Firestore pools requests
+      // across multiple clients.
+      for (let i = 0; i < 150; ++i) {
+        emptyResults[i] = new Deferred<void>();
+        documentResults[i] = new Deferred<void>();
+
+        unsubscribeCallbacks[i] = randomCol
+          .where('i', '>', i)
+          .onSnapshot(snapshot => {
+            if (snapshot.size === 0) {
+              emptyResults[i].resolve();
+            } else if (snapshot.size === 1) {
+              documentResults[i].resolve();
+            }
+          });
+      }
+
+      await Promise.all(emptyResults.map(d => d.promise));
+      ref.set({i: 1337});
+      await Promise.all(documentResults.map(d => d.promise));
+      unsubscribeCallbacks.forEach(c => c());
```

Comment:
This test verifies that all 150 listeners succeed but doesn't verify that everything has been properly released to the pool.

Is it possible to check that pool.size is 150 once the listeners are started and then get back to zero after?

Reply:
I worry that this test can succeed even if you remove the line that resolves the lifetime promise.